### ISSUE TITLE
upgrade all actions/cache

### ISSUE THREAD

**Issue by taojintianxia** [apache/shardingsphere#21400]
there are some cache are v2 and the rest are v3, need to use the same version.
take a look on the `restore-keys` of action cache, is this necessary ？

**Comment by taojintianxia** [comment]
there are some cache are v2 and the rest are v3, need to use the same version.
take a look on the `restore-keys` of action cache, is this necessary ？

**Comment by taojintianxia** [comment]
Followings are the v3 upgrade features :

  - Added support for caching from GHES 3.5.
  - Fixed download issue for files > 2GB during restore.
  - Updated the minimum runner version support from node 12 -> node 16.
  - Fixed avoiding empty cache save when no files are available for caching.
  - Fixed tar creation error while trying to create tar with path as ~/ home folder on ubuntu-latest.
  - Fixed zstd failing on amazon linux 2.0 runners.
  - Fixed cache not working with github workspace directory or current directory.
  - Fixed the download stuck problem by introducing a timeout of 1 hour for cache downloads.
  - Fix zstd not working for windows on gnu tar in issues.
  - Allowing users to provide a custom timeout as input for aborting download of a cache segment using an environment variable SEGMENT_DOWNLOAD_TIMEOUT_MIN. Default is 60 minutes.

**Comment by taojintianxia** [comment]
**Usage**
The cache action first searches for cache hits for key and restore-keys in the branch containing the workflow run. If there are no hits in the current branch, the cache action searches for key and restore-keys in the parent branch and upstream branches.

restore-keys allows you to specify a list of alternate restore keys to use when there is a cache miss on key. You can create multiple restore keys ordered from the most specific to least specific. The cache action searches the restore-keys in sequential order. When a key doesn't match directly, the action searches for keys prefixed with the restore key. If there are multiple partial matches for a restore key, the action returns the most recently created cache.

#### Example using multiple restore keys
```
restore-keys: |
  npm-feature-${{ hashFiles('package-lock.json') }}
  npm-feature-
  npm-
```

The runner evaluates the expressions, which resolve to these restore-keys:

```
restore-keys: |
  npm-feature-d5ea0750
  npm-feature-
  npm-
```

The restore key npm-feature- matches any key that starts with the string npm-feature-. For example, both of the keys npm-feature-fd3052de and npm-feature-a9b253ff match the restore key. The cache with the most recent creation date would be used. The keys in this example are searched in the following order:

1. npm-feature-d5ea0750 matches a specific hash.
2. npm-feature- matches cache keys prefixed with npm-feature-.
3. npm- matches any keys prefixed with npm-.

#### Example of search priority
1. Key npm-feature-d5ea0750 in the feature branch
2. Key npm-feature- in the feature branch
3. Key npm- in the feature branch
4. Key npm-feature-d5ea0750 in the main branch
5. Key npm-feature- in the main branch
6. Key npm- in the main branch
